src/lib/eventStack/eventStack.tsx

```diff
@@ -55,6 +54,25 @@ class EventStack {
   }
 }
 
-const instance = new EventStack()
+const eventStack = new EventStackInner()
+
+class EventStackSubscription {
+  public static empty() {
+    return new EventStackSubscription(() => {}, true)
+  }
+
+  public constructor(private _unsubscribe: Function, public readonly isEmpty: boolean = false) {}
 
-export default instance
+  public unsubscribe(): void {
+    this._unsubscribe()
+  }
+}
+
+export class EventStack {
```

Comment:
I am strongly against introducing any code that will lead to null checks on the client side - it should be safe to call `unsub` in the following way:

```js
class MyClassUsingEventStack {
  ...
  cleanup() {
    this.unsub()
  }
```

**Second thing** is that with just function being returned from subscription method (stored in `unsub` variable), it would be much harder to introduce changes to unsubscription logic - subscription's API won't be extensible, and each potential change to it will result in breaking API changes.

>Expose subscription function only from `eventStack` module
Agree, this is a good move, thanks!

Reply:
discussed with Roman and decided to stick with the approach he's introducing and revisit the EventStack API at a later time if needed